src/NotificationTargetCommonITILObject.php

```diff
@@ -1090,13 +1090,13 @@ public function addDataForTemplate($event, $options = [])
     /**
      * Get data from an item
      *
-     * @param CommonDBTM $item    Object instance
-     * @param array      $options Options
-     * @param boolean    $simple  (false by default)
+     * @param CommonITILObject  $item    Object instance
+     * @param array             $options Options
+     * @param boolean           $simple  (false by default)
      *
      * @return array
      **/
-    public function getDataForObject(CommonDBTM $item, array $options, $simple = false)
```

Comment:
While this is correct, I do not know if such a change should not be considered as a BC break :/

Reply:
It fixes these errors:
```
src/NotificationTargetCommonITILObject.php:1148:Call to an undefined method CommonDBTM::getStatus().
src/NotificationTargetCommonITILObject.php:1150:Call to an undefined method CommonDBTM::getUrgencyName().
src/NotificationTargetCommonITILObject.php:1151:Call to an undefined method CommonDBTM::getImpactName().
src/NotificationTargetCommonITILObject.php:1152:Call to an undefined method CommonDBTM::getPriorityName().
src/NotificationTargetCommonITILObject.php:1153:Call to an undefined method CommonDBTM::getActionTime().
src/NotificationTargetCommonITILObject.php:1172:Call to an undefined method CommonDBTM::countUsers().
src/NotificationTargetCommonITILObject.php:1174:Call to an undefined method CommonDBTM::getUsers().
src/NotificationTargetCommonITILObject.php:1203:Call to an undefined method CommonDBTM::countSuppliers().
src/NotificationTargetCommonITILObject.php:1205:Call to an undefined method CommonDBTM::getSuppliers().
src/NotificationTargetCommonITILObject.php:1238:Call to an undefined method CommonDBTM::countUsers().
src/NotificationTargetCommonITILObject.php:1240:Call to an undefined method CommonDBTM::getUsers().
src/NotificationTargetCommonITILObject.php:1261:Call to an undefined method CommonDBTM::countGroups().
src/NotificationTargetCommonITILObject.php:1263:Call to an undefined method CommonDBTM::getGroups().
src/NotificationTargetCommonITILObject.php:1276:Call to an undefined method CommonDBTM::countGroups().
src/NotificationTargetCommonITILObject.php:1278:Call to an undefined method CommonDBTM::getGroups().
src/NotificationTargetCommonITILObject.php:1291:Call to an undefined method CommonDBTM::countUsers().
src/NotificationTargetCommonITILObject.php:1293:Call to an undefined method CommonDBTM::getUsers().
src/NotificationTargetCommonITILObject.php:1311:Call to an undefined method CommonDBTM::countGroups().
src/NotificationTargetCommonITILObject.php:1313:Call to an undefined method CommonDBTM::getGroups().
src/NotificationTargetCommonITILObject.php:1422:Call to an undefined method CommonDBTM::getSolvedStatusArray().
src/NotificationTargetCommonITILObject.php:1423:Call to an undefined method CommonDBTM::getClosedStatusArray().
src/NotificationTargetCommonITILObject.php:1448:Call to an undefined method CommonDBTM::getAssociatedDocumentsCriteria().
src/NotificationTargetCommonITILObject.php:1524:Call to an undefined method CommonDBTM::getActionTime().
src/NotificationTargetCommonITILObject.php:1616:Call to an undefined method CommonDBTM::getTimelineItems().
```

Reply:
Agreed with @trasher

Reply:
Hum we didn't discuss that point last Thursday.
Is the 10.0bf target mandatory?
for next pr, as complexity increase, maybe we should target main branch, no?

Reply:
The method signature chzange does not seems required; changing only phpdoc is enough. This is not "complex"; method signature must be preserved; code must change as less as possible.

Reply:
Targeting 10.0 is **not** mandatory at all. But working on phpstan issues already fixed few issues.

Reply:
(and oh, I forget to mention; this is not mandatory we reach phpstan level 2 on bugfixes branch)

Reply:
In this case, should I revert the file to its original state? That’s probably the simplest solution.
_For your information, there are still around 200 level 2 errors, many of which are similar._

Reply:
Does phpdoc change cause an issue? If not, you can keep it.
Method signature change must be reverted, yes.

Reply:
Yes, I've just committed it.